extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -221,7 +224,9 @@ private static long write(
                         tableRowSet, columnSourceMap, dest, destOutputStream, writeInstructions, tableMeta,
                         tableInfoBuilder, metadataFileWriter);
                 // Given the transformation, do not use the original table's "definition" for writing
-                write(t, writeInstructions, parquetFileWriter, computedCache);
+                for (final Table tbl : writeInstructions.getRowGroupInfo().splitForRowGroups(t)) {
+                    write(tbl, writeInstructions, parquetFileWriter, computedCache);
+                }
```

Comment:
Since I think we should take control of all the impls, we can have whatever interface we want here. Allows for us to be more efficient, and have the implementations call write themselves; or, have the impls have an iterator that returns RowSets.

so, either the impls become something like:

```java
// ... instead of putting into an array, it will write it out itself
Table rowGroup = input.getSubTable(nextRows.toTracking());
write(rowGroup, writeInstructions, parquetFileWriter, computedCache)
// ...
```

or, this layer would do something like:

```java
final Iterator<RowSet> it = rowSets(writeInstructions.getRowGroupInfo());
while (it.hasNext()) {
  write(t.getSubTable(it.next()), writeInstructions, parquetFileWriter, computedCache);
}
```